### DISCUSSION TITLE
Expo + NextJs web ReferenceError: window is not defined

### DISCUSSION THREAD

**Discussion by cmaycumber** [software-mansion/react-native-reanimated#1940] [General]
I'm running into the following error: `ReferenceError: window is not defined on an app running expo and NextJS when Next is trying to use serverside rendering. 

The faulty file is here: `react-native-reanimated/lib/reanimated2/js-reanimated/JSReanimated.js:33:9`.

After changing the following line of code from:
```js
this.timeProvider = process.env.JEST_WORKER_ID
            ? { now: () => Date.now() }
            : window.performance;
```
to:
```js
this.timeProvider = process.env.JEST_WORKER_ID || typeof window === 'undefined'
            ? { now: () => Date.now() }
            : window.performance;
```

It solved my error.

My question is my configuration wrong or is this just a bug with serverside rendering? Does adding this check affect anything else?

**Comment by jgonet** [comment]
This is fixed in https://github.com/software-mansion/react-native-reanimated/commit/5a449ec21457bf7754fd9c618d0d89695b742fa0.

You can eject from managed workflow and use a [patch-package](https://www.npmjs.com/package/patch-package) to apply this commit for the time being or wait until the next expo release.

**Reply by cmaycumber** [reply]
Thanks much appreciated.
